### PR TITLE
Merging lnPs and file reading tools

### PR DESCRIPTION
**Pull request by lea-hagen**
This is basically a duplicate of #468, but with just the tools in that bulleted list, and not the star type probability code.  I've copied over that list here:

* `tools/subgridding_tools.py`: Added a `merge_lnp` function to combine lnPs across subgrids.  There's an optional `threshold` argument (like the one in `fit.py`) to eliminate lnPs much smaller than the maximum, which can help reduce file size.
* `tools/run/merge_files.py`: includes `merge_lnp` in the merging for subgrids
* `tools/read_beast_data.py`: Migrated over from `read_beast.py` in the MegaBEAST (BEAST-Fitting/megabeast#40).  It has functions to extract data from the files for the noise model, SED grid, and lnPs, and return them as dictionaries.  For the SED grid, if the parameter list is `None`, it will print out all of the parameters in the file (relevant to #433).  The code also has a function to get the parameter values associated with the lnPs.
* `tools/trim_many_via_obsdata.py`: If a trimmed file already exists, it will now skip to the next one in the input file.  (This isn't immediately related, but it came up while I was testing a start-to-finish run with the lnP merging.)

### CODE REVIEW

**Comment by karllark** [comment]
There is no update to the docs for these changes.  Is it because everything is already documented?  I feel like this PR has some changes that are not reflected in the current docs.  Please advise.

**Comment by lea-hagen** [comment]
I added some documentation about the new lnP merging tool and the reading tools.  Is there anything else I should address before this is merged?

File: beast/tools/read_beast_data.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
+"""
+Functions for interacting with the BEAST model
+"""
+
+# system imports
+from __future__ import (absolute_import, division, print_function)
```
Comment (karllark) [inline beast/tools/read_beast_data.py]: We don't support python 2, so the future imports can be removed.

File: beast/tools/run/merge_files.py [hunk 2]
```diff
@@ -5,8 +5,10 @@
 from beast.tools.run import create_filenames
 
 
+
 import datamodel
 import importlib
+importlib.reload(subgridding_tools)
```
Comment (karllark) [inline beast/tools/run/merge_files.py]: Remove.

File: beast/tools/trim_many_via_obsdata.py [hunk 3]
```diff
@@ -60,6 +60,11 @@
         sed_trimname = filebase + "_seds_trim.grid.hd5"
         noisemodel_trimname = filebase + "_noisemodel_trim.grid.hd5"
 
+        # if these already exist, then continue to the next set of files to trim
+        if os.path.isfile(sed_trimname) and os.path.isfile(noisemodel_trimname):
+            print("trimming already complete for "+sed_trimname)
+            continue
+
```
Comment (karllark) [inline beast/tools/trim_many_via_obsdata.py]: This is not related to the main topic of this PR.  Could be a separate PR.

Not sure that not just automatically overwriting the trim files isn't a better solution.  That is what would currently happen, right?  Or is the current behavior to crash?

Comment (lea-hagen) [reply beast/tools/trim_many_via_obsdata.py]: Happy to move it to a new PR.

The issue is that trimming can take a while, so skipping the file makes it run faster.

Comment (karllark) [reply beast/tools/trim_many_via_obsdata.py]: Please do.  I mainly want to make sure that this change does not mess up runs as right now I'm assuming that is does replace the file.  Would be a bummer if this silently changed to *not* updating the file and this was not the desired behavior.

**Comment by lea-hagen** [comment]
Just did a force push to undo merging master into this branch.  I'll do a rebase on my computer to resolve the documentation conflicts.

File: beast/tools/trim_many_via_obsdata.py (hunk 3 above)
Comment (lea-hagen) [reply beast/tools/trim_many_via_obsdata.py]: Opened a new PR for this (#501).
